CMakeLists.txt

```
@@ -58,17 +58,26 @@ target_include_directories(${PROJECT_NAME}
 	PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
 	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-	$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}> 
+	$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
 	${up-core-api_INCLUDE_DIR}
 	${protobuf_INCLUDE_DIR}
 	${spdlog_INCLUDE_DIR})
 
 set_property(TARGET ${PROJECT_NAME} PROPERTY POSITION_INDEPENDENT_CODE ON)
 
-target_link_libraries(${PROJECT_NAME} 
-	PRIVATE 
+target_compile_options(${PROJECT_NAME} PUBLIC
+	-Wall
+	-Wcast-qual
+	-Wuninitialized
+	-Wconversion
+	-pedantic
+	-Werror
+)
```

Add [`-Wswitch-enum`](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wswitch-enum). This will issue a warning for any switch statement on an enumerated type where one of the enumeration values is missing, _even if there is a `default:` case_.

```suggestion
target_compile_options(${PROJECT_NAME} PUBLIC
	-Wall
	-Wcast-qual
	-Wuninitialized
	-Wconversion
	-Wswitch-enum
	-pedantic
	-Werror
)
```